dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/rest/TelemetryClientFilter.java

```diff
@@ -107,10 +139,14 @@ public Void run() {
     /**
      * @return false if OpenTelemetry is disabled
      *         Indicated by instrumenter being set to null
+     * @return true when instrumenter is not null or if it is called during checkpoint
      */
     @Override
     public boolean isEnabled() {
-        return instrumenter != null;
+        if (!CheckpointPhase.getPhase().restored()) {
+            return true;
+        }         
+        return getInstrumenter() != null;  
     }
```

Comment:
I guess this makes sense. We're using `isEnabled` to decide whether or not to register the filter for performance reasons. If we're doing this sort of check before checkpoint, we don't know whether or not the filter will end up being enabled once the server is restored, so we have to register the filter in case it is enabled on restore.